src/Orleans.Core/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -112,6 +120,23 @@ public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceT
             return null;
         }
 
+        public async Task NextBatch(IBatchContainerBatch batchContainerBatch)
+        {
+            bool isRequestContextSet;
+            foreach (var batchContainer in batchContainerBatch.BatchContainers)
+            {
+                isRequestContextSet = batchContainer.ImportRequestContext();
+                foreach (var itemTuple in batchContainer.GetEvents<T>())
+                {
+                    await NextItem(itemTuple.Item1, itemTuple.Item2);
```

Comment:
I don't quite understand this. Why don't we deliver the whole batch by calling `OnNextBatchAsync`, but still deliver the batch item by item?

Reply:
Correct me if I'm wrong, but `OnNextBatchAsync` is actually not supported. If it was supported, then yes, I would be calling `OnNextBatchAsync` for each `IBatchContainer`.

Reply:
I see. I was thinking it might be because of that. But I was confused because I heard you guys are interested in implementing `OnNextBatchAsync` as well, and I'm not sure whether that is implemented in this pr or the next one. 

thanks for explaining.